cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java

```diff
@@ -55,10 +57,29 @@ static String getProposalSummary(GoalOptimizer.OptimizerResult result) {
       }
     }
     return String.format("%n%nThe optimization proposal has %d replica(%d MB) movements and %d leadership movements "
-                             + "based on the cluster model with %d recent snapshot windows and %.3f%% of the partitions "
-                             + "covered.", numReplicaMovements, dataToMove, numLeaderMovements,
-                         result.clusterModelStats().numSnapshotWindows(),
-                         result.clusterModelStats().monitoredPartitionsPercentage() * 100);
+            + "based on the cluster model with %d recent snapshot windows and %.3f%% of the partitions " + "covered.",
+            numReplicaMovements, dataToMove, numLeaderMovements, result.clusterModelStats().numSnapshotWindows(),
+            result.clusterModelStats().monitoredPartitionsPercentage() * 100);
   }
 
+  static Map<String, Object> getProposalSummaryInJson(GoalOptimizer.OptimizerResult result) {
+    int numReplicaMovements = 0;
+    int numLeaderMovements = 0;
+    long dataToMove = 0;
+    for (ExecutionProposal p : result.goalProposals()) {
+      if (!p.replicasToAdd().isEmpty() || !p.replicasToRemove().isEmpty()) {
+        numReplicaMovements++;
+        dataToMove += p.dataToMoveInMB();
+      } else {
+        numLeaderMovements++;
+      }
+    }
```

Comment:
This function duplicates the content of `getProposalSummary`. Can we use a common helper function for the shared code?

Reply:
moved all functions related with OptimizerResult from KafkaCruiseControlServletUtils class to OptimizerResult class.